R/sharpen.R

```diff
@@ -1,17 +1,63 @@
+# Install EBImmage
+#if (!requireNamespace("BiocManager", quietly = TRUE))
+#  install.packages("BiocManager")
+#BiocManager::install("EBImage", version = "3.8")
+
+library(EBImage)
 library(png)
+library(testit)
 
 #' Sharpen
 #'
+#' Sharpens an image with given intensity specification.
+#' This function will use the unsharp mask approach to sharpen images.
+#' Works by enhancing the edges of an image.
+#'
 #' @param input_img string, path of the input .png image
 #' @param intensity int, intensity of sharpness enhancement, between 0 and 10, defaults to 5
+#' @param display bool, if TRUE, output will be displayed. Defaults to FALSE.
 #' @param output_img string, path of the output .png image
 #'
 #' @return a png image at the specificed out path
 #' @export
 #'
 #' @examples
-#' sharpen("./input.png", 5, "./output.png")
+#' sharpen("./input.png", 5, FALSE, "./output.png")
+
+sharpen <- function(input_img, intensity=5, display=F, output_img=""){
+
+  # exception handling
+  # note that other eceptions are already handled by the PNG library functions readPNG and writePNG
+  testit::assert("Please provide a valid string for the input image path.", is.character(input_img))
+  testit::assert("Please provide a valid string for the output image path.", is.character(output_img))
+  testit::assert("Please provide an intensity value between 0 and 10.", intensity <= 10 & intensity >= 0)
+
+  # Read Img File
+  im = png::readPNG(input_img)
+
+  # scale pixel values from [0, 1] back to [0, 255]
+  im <- im * 255
+
+  # Get Gauss
+  imblur <- gblur(im, sigma=1)
```

Comment:
`gblur` is a function from an external package, ideally we have to explicitly reference this external package in R!
```suggestion
  imblur <- EBImage::gblur(im, sigma=1)
```
In addition, let's see if we can find a lighter weight library for blurring, I worry about the dependency management for including the EBImage package, which is not part of CRAN, I don't believe.